### ISSUE TITLE
write_csv option for tagged NA values

### ISSUE THREAD

**Issue by dusadrian** [tidyverse/readr#1193]
Tagged NA values, in `haven_labelled` objects have specific (and different meanings) than regular NA values, but all this information is lost when saving to a CSV file, all missing data being written as NA.

```r
library(haven)
library(readr)
path <- tempfile()
x <- data.frame(A = labelled(c(1, tagged_na('a'))))
write_csv(x, tempfile)

# A
# 1
# NA
```

Other software such as Stata export this dataset as:

```
# A
# 1
# .a
```

Would it be possible to add such an option for `write_csv()`?

**Comment by jimhester** [comment]
I don't think we want to include an option, however readr has a `output_column` generic, which it calls on columns before writing. If haven defined a `output_column.haven_labelled()` method this would be done automatically.

**Comment by jimhester** [comment]
e.g.

``` r
library(haven)
library(readr)
output_column.haven_labelled <- function(x) {
  format(x)
}

path <- tempfile()
x <- data.frame(A = labelled(c(1, tagged_na('a'))))
cat(format_csv(x))
#> A
#>     1
#> NA(a)
```

<sup>Created on 2021-04-12 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by jimhester** [comment]
I am going to move this issue over to the haven repository.
